File: BOJ/1260.py
from collections import deque

n, m, v = map(int, input().split())
graph = [[]*(n+1) for _ in range(n+1)]
d_visited = [0] * (n+1)
b_visited = [0] * (n+1)
for i in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for i in range(1, n+1):
    graph[i].sort()


def dfs(graph, V, d_visited):
    d_visited[V] = 1
    print(V, end=' ')

    for i in graph[V]:
        if d_visited[i] == 0:
            dfs(graph, i, d_visited)


def bfs(graph, start, b_visited):
    queue = deque([start])
    b_visited[start] = 1

    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for i in graph[v]:
            if b_visited[i] == 0:
                queue.append(i)
                b_visited[i] = 1


dfs(graph, v, d_visited)
print()
bfs(graph, v, b_visited)

File: BOJ/1260.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, M, V] = input[0].split(" ").map(Number);

const graph = Array.from({ length: N + 1 }, () => []);

for (let i = 1; i <= M; i++) {
  const [a, b] = input[i].split(" ").map(Number);
  graph[a].push(b);
  graph[b].push(a);
}

for (let i = 1; i <= N; i++) {
  graph[i].sort((a, b) => a - b);
}

const DFS_VISITED = new Array(N + 1).fill(false);
const BFS_VISTIED = new Array(N + 1).fill(false);
const DFS_RESULT = [];
const BFS_RESULT = [];

function dfs(v) {
  
  DFS_VISITED[v] = true;
  DFS_RESULT.push(v);

  for (const neighbor of graph[v]) {
    if (!DFS_VISITED[neighbor]) {
      dfs(neighbor);
    }
  }
}

function bfs(v) {
  const queue = [];
  BFS_VISTIED[v] = true;
  queue.push(v);

  while (queue.length > 0) {
    const node = queue.shift();
    BFS_RESULT.push(node);

    for (const neighbor of graph[node]) {
      if (!BFS_VISTIED[neighbor]) {
        BFS_VISTIED[neighbor] = true;
        queue.push(neighbor);
      }
    }
  }
}

dfs(V);
bfs(V);

console.log(DFS_RESULT.join(" "));
console.log(BFS_RESULT.join(" "));
